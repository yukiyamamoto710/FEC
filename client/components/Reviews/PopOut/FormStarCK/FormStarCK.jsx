import React from 'react';
import PropTypes from 'prop-types';
import Stars from '../../Stars/Stars';
import { Data } from '../../data';

const FormStarCK = (props) => {
  const {
    name,
    info,
    starClick,
  } = props;
  return (
    <div
      data-testid="formStarCK"
      className="topSmallContainer"
    >
      <div>
        { name }
      </div>
      <Stars
        className="formStar"
        rate={info}
        classNameForSize="formStar"
        starClick={starClick}
        name={String(name)}
      />
      <div
        data-testid="formStarCKText"
        className="formText"
      >
        {Data[name][info - 1]}
      </div>
    </div>
  );
};

FormStarCK.propTypes = {
  name: PropTypes.string,
  info: PropTypes.number,
  starClick: PropTypes.func,
};

FormStarCK.defaultProps = {
  name: '',
  info: 1,
  starClick: () => 1,
};
export default FormStarCK;
