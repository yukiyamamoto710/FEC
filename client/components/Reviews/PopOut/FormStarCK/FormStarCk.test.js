/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  screen,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import FormStarCK from './FormStarCK';

afterEach(cleanup);

const testName = 'Width';
const testFunc = jest.fn();

it('FormStarCK will render', () => {
  render(
    <FormStarCK
      name={testName}
      info={1}
      starClick={testFunc}
    />,
  );
  expect(screen.queryByText(testName)).toBeInTheDocument();
});

it('should have default starClick()', () => {
  expect(FormStarCK.defaultProps.starClick()).toBeDefined();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <FormStarCK
      name={testName}
      info={1}
      starClick={testFunc}
    />,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
