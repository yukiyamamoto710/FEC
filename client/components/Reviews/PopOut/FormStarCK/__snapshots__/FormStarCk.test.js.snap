// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="topSmallContainer"
  data-testid="formStarCK"
>
  <div>
    Width
  </div>
  <div
    className="ratingHead"
    data-testid="stars"
  >
    <div>
      <div
        className="formStar fullStar"
        data-testid="button0"
        id="Width 1"
        label="fullStar"
        name="Width"
        onClick={[MockFunction]}
        onKeyPress={[MockFunction]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="formStar emptyStar"
        data-testid="button1"
        id="Width 2"
        label="emptyStar"
        name="Width"
        onClick={[MockFunction]}
        onKeyPress={[MockFunction]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="formStar emptyStar"
        data-testid="button2"
        id="Width 3"
        label="emptyStar"
        name="Width"
        onClick={[MockFunction]}
        onKeyPress={[MockFunction]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="formStar emptyStar"
        data-testid="button3"
        id="Width 4"
        label="emptyStar"
        name="Width"
        onClick={[MockFunction]}
        onKeyPress={[MockFunction]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="formStar emptyStar"
        data-testid="button4"
        id="Width 5"
        label="emptyStar"
        name="Width"
        onClick={[MockFunction]}
        onKeyPress={[MockFunction]}
        role="button"
        tabIndex={0}
      />
    </div>
  </div>
  <div
    className="formText"
    data-testid="formStarCKText"
  >
    Too narrow
  </div>
</div>
`;
